### PR TITLE
chore: tests migration to assertj (KotlinStdlibCompatUtilsTest, ModelAutolinkingDependenciesJsonTest, PathUtilsTest, DependencyUtilsTest, ProjectUtilsTest)

### PR DESCRIPTION
**Pull request by kachmashk**
## Summary:

https://github.com/facebook/react-native/issues/45596

## Changelog:

Migrated to AssertJ within files:
- `KotlinStdlibCompatUtilsTest.kt`,
- `ModelAutolinkingDependenciesJsonTest.kt`,
- `PathUtilsTest.kt`,
- `DependencyUtilsTest.kt`,
- `ProjectUtilsTest.kt`.

[INTERNAL] [CHANGED] - Migrated `KotlinStdlibCompatUtilsTest`, `ModelAutolinkingDependenciesJsonTest`, `PathUtilsTest`, `DependencyUtilsTest`, `ProjectUtilsTest` from `junit.Assert` to `assertj.core.api.Assertions`.

## Test Plan

### CODE REVIEW

**Comment by kachmashk** [comment]
Hey @cortinico, cannot add you as a reviewer so @ you here.

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt [hunk 1]
```diff
@@ -184,8 +184,7 @@ class DependencyUtilsTest {
 
     // We need to make sure we have Maven Central defined twice, one by the library,
     // and another is the override by RNGP.
-    assertEquals(
-        2,
+    assertThat(2).isEqualTo(
         libProject.repositories.count { it is MavenArtifactRepository && it.url == repositoryURI })
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt:188]: Can you invert this statement? `assertThat(...).isEqualTo(2)`

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt [hunk 2]
```diff
@@ -301,7 +300,7 @@ class DependencyUtilsTest {
 
     val versionString = readVersionAndGroupStrings(propertiesFile).first
 
-    assertEquals("1000.0.0", versionString)
+    assertThat("1000.0.0").isEqualTo(versionString)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt:303]: can you invert this

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt [hunk 3]
```diff
@@ -318,7 +317,7 @@ class DependencyUtilsTest {
 
     val versionString = readVersionAndGroupStrings(propertiesFile).first
 
-    assertEquals("0.0.0-20221101-2019-cfe811ab1-SNAPSHOT", versionString)
+    assertThat("0.0.0-20221101-2019-cfe811ab1-SNAPSHOT").isEqualTo(versionString)
```
Comment (cortinico) [inline packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt:320]: same

**Review by cortinico — CHANGES_REQUESTED** [review]
Can you invert the equal assertion so that is:

`assertThat(actual).isEqualTo(expected)`

You have this in several tests

**Review by cortinico — CHANGES_REQUESTED** [review]
Can you invert the equal assertion so that is:

`assertThat(actual).isEqualTo(expected)`

You have this in several tests

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt (hunk 1 above)
Comment (kachmashk) [reply packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt]: Fixed

File: packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt (hunk 3 above)
Comment (kachmashk) [reply packages/gradle-plugin/react-native-gradle-plugin/src/test/kotlin/com/facebook/react/utils/DependencyUtilsTest.kt]: Fixed

**Review by cortinico — APPROVED** [review]

**Comment by kachmashk** [comment]
Hey @cortinico, how could I fix the `Facebook Internal - Linter`? I cannot see what went wrong as it's only for Meta's employees.

**Comment by cortinico** [comment]
> Hey @cortinico, how could I fix the `Facebook Internal - Linter`? I cannot see what went wrong as it's only for Meta's employees.

I fixed it for you. The status doesn't update automatically, but the final commit that will land will be properly formatted 👍

**Pull request closed without merging**